umbral/curvebn.py

```diff
@@ -85,19 +85,34 @@ def hash_to_bn(cls, *crypto_items, params=None):
                     raise TypeError("{} is not acceptable type, received {}".format(item, type(item)))
                 item_bytes = item
             blake2b.update(item_bytes)
+        
+        blake2b.update(_iteration.to_bytes(4, byteorder='big'))
 
-        i = 0
-        h = 0
-        while h < params.CURVE_MINVAL_HASH_512:
-            blake2b_i = blake2b.copy()
-            blake2b_i.update(i.to_bytes(params.CURVE_KEY_SIZE_BYTES, 'big'))
-            hash_digest = blake2b_i.finalize()
-            h = int.from_bytes(hash_digest, byteorder='big', signed=False)
-            i += 1
-
-        hash_bn = h % int(params.order)
-        res = cls.from_int(hash_bn, params.curve)
-        return res
+        hash_digest = blake2b.finalize()
+        hash_digest = int.from_bytes(hash_digest, byteorder='big', signed=False)
+
+        curve_nid = backend._elliptic_curve_to_nid(params.curve)
+        order = openssl._get_ec_order_by_curve_nid(curve_nid)
+
+        try:
```

Comment:
Is this `try` block too widely scoped - which line actually raises the `ValueError`?  Perhaps `try/except/else` is useful here.

Reply:
Yes, you're right. It's actually the first line. I like having everything in the try since it reads better in case there's not exception, which happens with overwhelming probability. But I recognize it's not a good practice, so I'll change it for a `try/except/else`.